src/lib/dom-api.html

```diff
@@ -593,6 +593,61 @@
 
     };
 
+    Object.defineProperties(DomApi.prototype, {
+      activeElement: {
+        configurable: true,
+        get: function() {
+          var ownerDocument = this.node.ownerDocument;
+
+          // Shadow DOM
+          if (Polymer.Settings.useShadow) {
+            if (ownerDocument || !window.ShadowDOMPolyfill) {
+              return this.node.activeElement;
+            } else {
+              return window.ShadowDOMPolyfill.wrapIfNeeded(this.node).activeElement;
+            }
+          }
+
+          // Shady DOM
+
+          var trueDocument = ownerDocument || this.node;
+
+          // Not a document or shady root.
+          if (ownerDocument && !this.node._isShadyRoot) {
+            return this.node.activeElement;
+          }
+
+          var active = trueDocument.activeElement;
+          if (!active) return null;
+
+          // `this.node` is either a document or shady root; iterate upwards
+          // to find the active element's most shallow host (relative to this
+          // node or its containing document).
+          var activeRoot = Polymer.dom(active).getOwnerRoot();
```

Comment:
How would contains make this faster? Do you mean skip `getOwnerRoot` when the active element is actually in this shady root (rather than nested deeper)?